### PR TITLE
feat(esl-popup): add handler for refresh event

### PR DESCRIPTION
**Pull request by dshovchko**

### CODE REVIEW

File: src/modules/esl-popup/core/esl-popup.ts [hunk 1]
```diff
@@ -14,11 +15,13 @@ import {ESLIntersectionTarget, ESLIntersectionEvent} from '../../esl-event-liste
 import {calcPopupPosition, isMajorAxisHorizontal} from './esl-popup-position';
 import {ESLPopupPlaceholder} from './esl-popup-placeholder';
 
+import type {ESLListenerCriteria, ESLListenerDescriptorExt} from '../../esl-event-listener/core';
 import type {ESLToggleableActionParams} from '../../esl-toggleable/core';
 import type {PositionType, IntersectionRatioRect} from './esl-popup-position';
 
 const INTERSECTION_LIMIT_FOR_ADJACENT_AXIS = 0.7;
 const DEFAULT_OFFSET_ARROW = 50;
+const OBSERVER_DESCRIPTOR: ESLListenerDescriptorExt = {auto: false, group: 'observer'};
```
[the remark this answers is not in the document]
Comment (dshovchko) [reply src/modules/esl-popup/core/esl-popup.ts:24]: Agree. Look like should be rewritten to regular style
But descriptors are very big :(

File: src/modules/esl-popup/core/esl-popup.ts [hunk 2]
```diff
@@ -331,23 +331,31 @@ export class ESLPopup extends ESLToggleable {
...
-  @listen({auto: false, event: 'resize', target: window})
+  @listen({auto: false, group: 'observer', event: 'resize', target: window})
   @decorate(rafDecorator)
   protected _onResize(): void {
     this._updatePosition();
   }
 
+  @listen({auto: false, group: 'observer', event: ($popup: ESLPopup) => $popup.REFRESH_EVENT, target: window})
+  protected _onRefresh(e: Event): void {
+    const {target} = e;
```
Comment (ala-n) [inline src/modules/esl-popup/core/esl-popup.ts:353]:
```suggestion
  protected _onRefresh({target}: Event): void {
```

File: src/modules/esl-popup/core/esl-popup.ts [hunk 3]
```diff
@@ -331,23 +331,31 @@ export class ESLPopup extends ESLToggleable {
...
+    if (!isElement(target)) return;
+    const {activator, $container} = this;
+    if ($container === target || this.contains(target) || activator?.contains(target)) this._updatePosition();
```
Comment (ala-n) [inline src/modules/esl-popup/core/esl-popup.ts:356]: In our case:
```
<div resizable>
  <div class = "something" >
      <esl-trigger>...<esl-trigger>
      <esl-popup-placeholder>...</esl-popup-placeholder>  
   </div>
   <div class="hey not only popup interested in esl:refresh"  
     ...
    </div>
 </div> 
 ```

so as a an `div[resizble]` maintainers we do not care about content we dispatch `esl:refres` on   `div[resizble]`


I'd suggest using our standard check for the third condition
```suggestion
    const {activator, $container} = this;
    if ($container === target || this.contains(target) || isRelativeNode(this.activator, target)) this._updatePosition();
```

**Review by NastaLeo — APPROVED** [review]

**Review by abarmina — APPROVED** [review]

File: src/modules/esl-popup/core/esl-popup.ts [hunk 4]
```diff
@@ -331,23 +327,30 @@ export class ESLPopup extends ESLToggleable {
...
   @decorate(rafDecorator)
   protected _onResize(): void {
     this._updatePosition();
   }
 
+  @listen({auto: false, group: 'observer', event: ($popup: ESLPopup) => $popup.REFRESH_EVENT, target: window})
+  protected _onRefresh({target}: Event): void {
+    if (!isElement(target)) return;
+    const {activator, $container} = this;
+    if ($container === target || this.contains(target) || activator && isRelativeNode(activator, target)) this._updatePosition();
```
Comment (ala-n) [inline src/modules/esl-popup/core/esl-popup.ts:351]:
```suggestion
    if ($container === target || this.contains(target) || isRelativeNode(activator, target)) this._updatePosition();
```
isRelativeNode is null safe

**Review by ala-n — APPROVED** [review]

**Review by fshovchko — APPROVED** [review]

File: src/modules/esl-popup/core/esl-popup.ts (hunk 4 above)
Comment (dshovchko) [reply src/modules/esl-popup/core/esl-popup.ts:351]: but

Comment (dshovchko) [reply src/modules/esl-popup/core/esl-popup.ts:351]: but `public get activator(): HTMLElement | null | undefined`

Here is the working code
```suggestion
    if ($container === target || this.contains(target) || isRelativeNode(activator ?? null, target)) this._updatePosition();
```

**Review by ala-n — APPROVED** [review]

**Review by yadamskaya — APPROVED** [review]

**Review by Natalie-Smirnova — APPROVED** [review]

**Pull request merged**